Tools/CISupport/ews-build/events.py

```diff
@@ -89,25 +95,41 @@ def __init__(self, master_hostname, type_prefix='', name='Events'):
         self.type_prefix = type_prefix
         self.master_hostname = master_hostname
 
-    def sendData(self, data):
+    def sendDataToEWS(self, data):
         if os.getenv('EWS_API_KEY', None):
             data['EWS_API_KEY'] = os.getenv('EWS_API_KEY')
         agent = Agent(reactor)
         body = JSONProducer(data)
 
         agent.request(b'POST', self.EVENT_SERVER_ENDPOINT, Headers({'Content-Type': ['application/json']}), body)
 
+    def sendDataToGitHub(self, repository, sha, data):
+        username, access_token = GitHub.credentials()
+
+        data['description'] = data.get('description', '')
+        if len(data['description']) > self.MAX_GITHUB_DESCRIPTION:
+            data['description'] = '{}...'.format(data['description'][:self.MAX_GITHUB_DESCRIPTION - 3])
+
+        agent = Agent(reactor)
+        body = JSONProducer(data)
+        d = agent.request(b'POST', '{}/statuses/{}'.format(GitHub.api_url(repository), sha).encode('utf-8'), Headers({
```

Comment:
How exactly would we do that though? We have at least two different URL paths that we're using at the moment, there will likely be 3 or 4 by the time we're done with GitHub stuff in EWS. I suppose we could pass a "path" argument to the api_url function, that would look like this:
`d = agent.request(b'POST', GitHub.api_url(repository, 'statuses/{}'.format(sha)), sha).encode('utf-8'))...`
which doesn't seem any better, possibly more confusing because you would need to consult the function to see what the extra argument is doing.